src/Associativity.cpp

```diff
@@ -273,7 +273,7 @@ void add_transitive_dependencies(vector<set<int>> &dependencies) {
                     for (const auto &idx : dependencies[j]) {
                         if (dependencies[i].count(idx) == 0) {
                             dependencies[i].insert(idx);
-                            change = false;
+                            change = true;
```

Comment:
I am not familiar with the code, but it does look like a legit bug.

Reply:
Agreed, but presumably the change will affect behavior somewhere. If it doesn't affect an existing test, we need a new one added to verify.

Reply:
It would have to be some very subtle almost-associative update definition that was previously incorrectly allowed. I think it would be quite challenging to find one.